### DISCUSSION TITLE
EKS Upgrade tool by AWS

### DISCUSSION THREAD

**Discussion by onabison** [aws-samples/eks-cluster-upgrade#67] [General]
Hi,

I have been trying to use the provided AWS EKS Upgrade tool:
https://github.com/aws-samples/eks-cluster-upgrade

While using it I have noticed a few things that I'm not sure why they are occurring:

1. The add-on seems to be downgrading rather than upgrading, here's the output:
`INFO:eksupgrade.src.k8s_client:coredns-79fffc5cf7-rtpnt Current Version = v1.8.7-eksbuild.3 Updating to = v1.8.7-eksbuild.2`

2. I am curious to understand how does the tool determine what is the latest available version for the respective add-on. It seems there is a much newer version for each of the add-on's I have on my cluster, yet, it manages to downgrade. Even if it does find an upgrade version, it only makes a small (usually minor version) upgrade. here's an image that shows the current version the upgrade tool selected during the upgrade process and VS the latest version that is available for that particular add-on.

![image](https://user-images.githubusercontent.com/35974508/220989959-bd23800e-266c-4392-9609-0b4b46cf6958.png)

**Comment by bryantbiggs** [comment]
Currently the logic is choosing to use the default addon version for the given Kubernetes version. However this is a great callout and we should make some changes to address. Specifically, we can leave the current default behavior as is (use the default addon version for the given Kubernetes version), but allow users to specify a flag  to use the latest addon version

**Reply by onabison** [reply]
Yet another suggestion:
If you are going to allow users to select the version they want to upgrade to (which is am amazing feature), could it be possible to re-run the same command (meaning not trying to upgrade the control plane) but it will still allow you to select add-on version, at which point you would be able to revert back to the lower version for the same Cluster version? 

I can understand if for example you upgraded from 1.23 to 1.24 and re-run the command:
`eksupgrade lab-poc-eks-cluster 1.24 us-east-1`

You won't be able to select add-on versions for 1.23. 

Please let me know if what I just said makes sense...

Thanks!
